### PR TITLE
Remove unused header iomgr_posix.h

### PR DESCRIPTION
**Pull request by drfloob**
This header file only served to include iomgr_internal.h, which most
dependent code already includes.

### CODE REVIEW

File: src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py [hunk 1]
```diff
@@ -150,8 +150,8 @@ def GetClientStats(
         return response
 
     def GetClientAccumulatedStats(
-            self, request: messages_pb2.LoadBalancerAccumulatedStatsRequest,
-            context: grpc.ServicerContext
+        self, request: messages_pb2.LoadBalancerAccumulatedStatsRequest,
```
Comment (markdroth) [inline src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py:153]: The changes in this file don't seem to be related to the rest of the PR.  If this is something that's broken in master, let's fix that in a separate PR before merging this one.

**Review by markdroth — APPROVED** [review]
This looks good, but please do split the yapf change out into a separate fix instead of merging it with this PR.  Thanks!

File: src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py (hunk 1 above)
Comment (drfloob) [reply src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py]: Good idea, I've removed it from this PR. From the yapf upgrade yesterday, I'm guessing we may expect more formatting changes as we go along.

**Review by markdroth — APPROVED** [review]

File: src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py (hunk 1 above)
Comment (markdroth) [reply src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py]: There shouldn't be more changes needed.  This was probably caused by two PRs being in flight at the same time -- the yapf change and another one that added a new file.  It shouldn't happen on an ongoing basis.

Comment (drfloob) [reply src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py]: Jan fixed the formatting issue in e2b17027

**Comment by drfloob** [comment]
Known issue: #23822 

See https://source.cloud.google.com/results/invocations/66ca5e0f-fa5d-4c37-bab8-1f563ccfd6d4/targets/github%2Fgrpc%2Frun_tests%2Fruby_linux_dbg_native%2Ftools%2Frun_tests%2Fhelper_scripts%2Frun_ruby.sh/tests
